### DISCUSSION TITLE
How does PUB/SUB behave in Java?

### DISCUSSION THREAD

**Discussion by thxrben** [xetorthio/jedis#3569] [Q&A] [answered]
In the REDIS-Docs, it is said that once you subscribed, you should not send any other commands (except unsub/quit/etc.).
In the Github Wiki under Advanced Usages - Pub / Sub you create a listener.

Does this mean that the Jedis library is capable of operating as usual and firing the listener if needed?
Do I need a second JedisCluster instance in order to subscribe and still use operations on e.g. hashes?

**Comment by sazzad16** [comment, marked as answer]
> Does this mean that the Jedis library is capable of operating as usual and firing the listener if needed?

No.

> Do I need a second JedisCluster instance in order to subscribe and still use operations on e.g. hashes?

IMO, that's better. Because if you subscribe (multiple times), some connections would be busy in subscription. This means you would have less number of resources available to server your other operations.

**Reply by thxrben** [reply]
Alright, thank you very much!

**Reply by thxrben** [reply]
Short follow-up question: Can i just call .clone() on the initial connection (JedisCluster) or do I have to create a brand new object?

**Reply by sazzad16** [reply]
a brand new object
